### PR TITLE
Remove FakeAddressSet from network policy testing

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
in an effort to remove the FakeAddressSet code from testing this commit remove FakeAddressSet from network policy testing

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: go-controller/pkg/ovn/policy_test.go [hunk 1]
```diff
@@ -2311,3 +2372,9 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Low-Level Operations", func() {
 		gomega.Expect(gp.delNamespaceAddressSet(four.GetObjectID(libovsdbops.ObjectNameKey))).To(gomega.BeFalse())
 	})
 })
+
+func buildNetworkPolicyAddressSets(namespaceName string, peer knet.NetworkPolicyPeer, ips ...string) (*nbdb.AddressSet, *nbdb.AddressSet) {
```
Comment (JacobTanenbaum) [inline go-controller/pkg/ovn/policy_test.go:2376]: I want to change all the build address set functions to this format taking variadic arguments this is just the first one to get this treatment

Comment (maiqueb) [reply go-controller/pkg/ovn/policy_test.go:2376]: But is the PR ready for review ? Will you be making those changes in this PR ?

Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go:2376]: my plan was to make those in future PRs.

Comment (maiqueb) [reply go-controller/pkg/ovn/policy_test.go:2376]: I'll review this one then ASAP. Thank you.

File: go-controller/pkg/ovn/policy_test.go [hunk 2]
```diff
@@ -1795,14 +1833,18 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
...
+				//eventuallyExpectAddressSetsWithIP(fakeOvn, networkPolicy.Spec.Egress[0].To[0],
+				//	networkPolicy.Namespace, nPodTest.podIP)
 
 				expectedData := getNamespaceWithSinglePolicyExpectedData(
 					newNetpolDataParams(networkPolicy),
 					initialDB.NBData)
+				netpolASv4, _ := buildNetworkPolicyAddressSets(namespace1.Name, networkPolicy.Spec.Egress[0].To[0], nPodTest.podIP)
+				expectedData = append(expectedData, netpolASv4)
+				namespace1AddressSetv4, _ := buildNamespaceAddressSets(namespace1.Name, nil)
+				expectedData = append(expectedData, namespace1AddressSetv4)
```
Comment (maiqueb) [inline go-controller/pkg/ovn/policy_test.go:1847]: nit

```suggestion
				netpolASv4, _ := buildNetworkPolicyAddressSets(namespace1.Name, networkPolicy.Spec.Egress[0].To[0], nPodTest.podIP)
				namespace1AddressSetv4, _ := buildNamespaceAddressSets(namespace1.Name, nil)
				expectedData = append(expectedData, netpolASv4, namespace1AddressSetv4)
```

**Review by maiqueb — COMMENTED** [review]
Code-wise, I'm OK with it. Makes sense IMHO.

I am just asking for code to be removed, rather than commented out.

Also, please explain the motivation for this commit / PR both in the commit message, and in the PR description. 

It isn't exactly obvious to me why getting rid of the fake address stuff in fake OVN is good. 

Thanks for this PR !

File: go-controller/pkg/ovn/policy_test.go [hunk 3]
```diff
@@ -748,14 +743,16 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
 			app.Action = func(ctx *cli.Context) error {
 				namespace1 := *newNamespace(namespaceName1)
 				namespace2 := *newNamespace(namespaceName2)
+				namespace1AddressSetv4, _ := buildNamespaceAddressSets(namespace1.Name, nil)
+				namespace2AddressSetv4, _ := buildNamespaceAddressSets(namespace2.Name, nil)
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go:747]: that was what my plan was for my follow up, I want to start a file that is just for test helper functions

[the remark this answers is not in the document]
Comment (maiqueb) [reply go-controller/pkg/ovn/policy_test.go:747]: I still think that makes sense as a separate commit in this PR, but I won't block on that. Thank you.

File: go-controller/pkg/ovn/policy_test.go [hunk 4]
```diff
@@ -1981,14 +2040,16 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
 				_, err := fakeOvn.fakeClient.KubeClient.NetworkingV1().NetworkPolicies(networkPolicy.Namespace).
 					Get(context.TODO(), networkPolicy.Name, metav1.GetOptions{})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				fakeOvn.asf.ExpectAddressSetWithAddresses(namespaceName1, []string{nPodTest.podIP})
+				//				fakeOvn.asf.ExpectAddressSetWithAddresses(namespaceName1, []string{nPodTest.podIP})
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: done

File: go-controller/pkg/ovn/policy_test.go [hunk 5]
```diff
@@ -1795,14 +1833,18 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
 					Get(context.TODO(), networkPolicy.Name, metav1.GetOptions{})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
 				// namespaced address set won't have hostNet pod ip
-				fakeOvn.asf.EventuallyExpectEmptyAddressSetExist(namespaceName1)
+				//fakeOvn.asf.EventuallyExpectEmptyAddressSetExist(namespaceName1)
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: yes, done

File: go-controller/pkg/ovn/policy_test.go [hunk 6]
```diff
@@ -1795,14 +1833,18 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
...
 				// netpol peer address set will
-				eventuallyExpectAddressSetsWithIP(fakeOvn, networkPolicy.Spec.Egress[0].To[0],
-					networkPolicy.Namespace, nPodTest.podIP)
+				//eventuallyExpectAddressSetsWithIP(fakeOvn, networkPolicy.Spec.Egress[0].To[0],
+				//	networkPolicy.Namespace, nPodTest.podIP)
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: removed

File: go-controller/pkg/ovn/policy_test.go [hunk 7]
```diff
@@ -1699,24 +1735,26 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
...
+				as.AddAddresses([]string{"10.244.0.2"})
 
 				// create networkPolicy, check db
 				_, err = fakeOvn.fakeClient.KubeClient.NetworkingV1().NetworkPolicies(networkPolicy.Namespace).
 					Get(context.TODO(), networkPolicy.Name, metav1.GetOptions{})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				fakeOvn.asf.ExpectAddressSetWithAddresses(namespaceName1, []string{nPodTest.podIP})
-				fakeOvn.asf.ExpectAddressSetWithAddresses(hostNamespaceName, []string{"10.244.0.2"})
+				//fakeOvn.asf.ExpectAddressSetWithAddresses(namespaceName1, []string{nPodTest.podIP})
+				//fakeOvn.asf.ExpectAddressSetWithAddresses(hostNamespaceName, []string{"10.244.0.2"})
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: removed

File: go-controller/pkg/ovn/policy_test.go [hunk 8]
```diff
@@ -1921,12 +1970,20 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
 					getPolicyData(newNetpolDataParams(initialDenyAllPolicy).
 						withAllowLogSeverity(updatedLogSeverity))...)
 				for i := range networkPolicies {
+					if networkPolicies[i].Spec.Egress != nil && networkPolicies[i].Spec.Egress[0].To[0].PodSelector != nil {
+						netpolASv4, _ := buildNetworkPolicyAddressSets(networkPolicies[i].Namespace, networkPolicies[i].Spec.Egress[0].To[0])
+						expectedData = append(expectedData, netpolASv4)
+					} else if networkPolicies[i].Spec.Ingress != nil && networkPolicies[i].Spec.Ingress[0].From[0].PodSelector != nil {
+						netpolASv4, _ := buildNetworkPolicyAddressSets(networkPolicies[i].Namespace, networkPolicies[i].Spec.Ingress[0].From[0])
+						expectedData = append(expectedData, netpolASv4)
+					}
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: done

**Review by maiqueb — APPROVED** [review]

**Comment by JacobTanenbaum** [comment]
hard coded network for test is not correct: 

  [FAILED] failed to create secondary network "secondary-network" with subnet(s) 2001:db8:abcd:1234:c000::/64: failed to run "docker network create --internal --driver bridge secondary-network --subnet 2001:db8:abcd:1234:c000::/64 --ipv6": exit status 1 (Error response from daemon: invalid network config:
  invalid subnet 2001:db8:abcd:1234:c000::/64: it should be 2001:db8:abcd:1234::/64
  )

**Comment by tssurya** [comment]
> hard coded network for test is not correct:
> 
> [FAILED] failed to create secondary network "secondary-network" with subnet(s) 2001:db8:abcd:1234:c000::/64: failed to run "docker network create --internal --driver bridge secondary-network --subnet 2001:db8:abcd:1234:c000::/64 --ipv6": exit status 1 (Error response from daemon: invalid network config: invalid subnet 2001:db8:abcd:1234:c000::/64: it should be 2001:db8:abcd:1234::/64 )

@kyrtapz has a fix here: https://github.com/ovn-org/ovn-kubernetes/pull/4450 for fixing this.

File: go-controller/pkg/ovn/policy_test.go [hunk 9]
```diff
@@ -2311,3 +2356,43 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Low-Level Operations", func() {
 		gomega.Expect(gp.delNamespaceAddressSet(four.GetObjectID(libovsdbops.ObjectNameKey))).To(gomega.BeFalse())
 	})
 })
+
+// buildNetworkPolicyPeerAddressSet builds the addresssets for the networkpolicy peer provided
+func buildNetworkPolicyPeerAddressSets(namespaceName string, peer knet.NetworkPolicyPeer, ips ...string) (*nbdb.AddressSet, *nbdb.AddressSet) {
```
Comment (tssurya) [inline go-controller/pkg/ovn/policy_test.go:2361]: I see potential to use this also in `ginkgo.It("creates one address set for multiple users with the same selector", func() {` that test

File: go-controller/pkg/ovn/policy_test.go [hunk 10]
```diff
@@ -488,11 +489,6 @@ func getHairpinningACLsV4AndPortGroup() []libovsdbtest.TestData {
 	return []libovsdbtest.TestData{egressACL, ingressACL, clusterPortGroup}
 }
 
-func eventuallyExpectNoAddressSets(fakeOvn *FakeOVN, peer knet.NetworkPolicyPeer, namespace string) {
-	dbIDs := getPodSelectorAddrSetDbIDs(getPodSelectorKey(peer.PodSelector, peer.NamespaceSelector, namespace), DefaultNetworkControllerName)
-	fakeOvn.asf.EventuallyExpectNoAddressSet(dbIDs)
-}
-
 func eventuallyExpectAddressSetsWithIP(fakeOvn *FakeOVN, peer knet.NetworkPolicyPeer, namespace, ip string) {
```
Comment (tssurya) [inline go-controller/pkg/ovn/policy_test.go:492]: what about the following two `eventuallyExpectAddressSetsWithIP` and `eventuallyExpectEmptyAddressSetsExist` functions that seem to use fakeOVN.asf, we shouldn't be using these anymore right? given we want to remove fake address set factory...

File: go-controller/pkg/ovn/policy_test.go [hunk 11]
```diff
@@ -2311,3 +2356,43 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Low-Level Operations", func() {
 		gomega.Expect(gp.delNamespaceAddressSet(four.GetObjectID(libovsdbops.ObjectNameKey))).To(gomega.BeFalse())
 	})
 })
+
```
Comment (tssurya) [inline go-controller/pkg/ovn/policy_test.go:2359]: nit: FWIW in EIP test we put UTILs at the bottom, but in this file they seem to be at the top, I won't block on this accord but in the future putting it here could lead to duplication of utils given ppl might look  for a set at the start and then not notice these below here..I'd keep them consistent together in one place (near the existing address set utils)

**Review by tssurya — CHANGES_REQUESTED** [review]
did not do in depth review given there is already a +1 but have some questions..

File: go-controller/pkg/ovn/policy_test.go (hunk 9 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go:2347]: can probably also be used there but that would involve removing the FakeAddressSet from that test file next.

File: go-controller/pkg/ovn/policy_test.go (hunk 10 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: moved these two functions to where they are still used pkg/ovn/pod_selector_address_set_test.go

File: go-controller/pkg/ovn/policy_test.go (hunk 9 above)
Comment (tssurya) [reply go-controller/pkg/ovn/policy_test.go:2347]: oh that's another PR?

File: go-controller/pkg/ovn/policy_test.go (hunk 11 above)
Comment (tssurya) [reply go-controller/pkg/ovn/policy_test.go:2345]: @JacobTanenbaum ^ PTAL

**Review by tssurya — DISMISSED** [review]
ptal the nit to move up the utils? outside of that lgtm

File: go-controller/pkg/ovn/policy_test.go (hunk 9 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go:2347]: that will be another PR, I am going one at a time to keep the changes manageable

File: go-controller/pkg/ovn/policy_test.go (hunk 11 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/policy_test.go]: moved the functions with the other test utils

**Comment by tssurya** [comment]
shard failure is already tracked here: https://github.com/ovn-org/ovn-kubernetes/issues/4340

**Review by tssurya — APPROVED** [review]

**Pull request merged**